frontend/src/components/common/ChangePassword/ChangePasswordModal.tsx

```diff
@@ -0,0 +1,264 @@
+ */
+const ChangePasswordModal = (
+  props: ChangePasswordModalProps,
+): React.ReactElement => {
+  const { authenticatedUser } = useContext(AuthContext);
+
+  const { isOpen, onClose } = props;
+  const [oldPassword, setOldPassword] = React.useState<string>("");
+  const [newPassword, setNewPassword] = React.useState<string>("");
+  const [confirmPassword, setConfirmPassword] = React.useState<string>("");
+
+  const [
+    oldPasswordFeedback,
+    setOldPasswordFeedback,
+  ] = React.useState<PasswordFeedback>({ isInvalid: false, errorMessage: "" });
+  const [
+    newPasswordFeedback,
+    setNewPasswordFeedback,
+  ] = React.useState<PasswordFeedback>({ isInvalid: false, errorMessage: "" });
+  const [
+    confirmPasswordFeedback,
+    setConfirmPasswordFeedback,
+  ] = React.useState<PasswordFeedback>({ isInvalid: false, errorMessage: "" });
+
+  const [feedback, setFeedback] = React.useState<string>("");
+  const [submitted, setSubmitted] = React.useState<boolean>(false);
+  const [error, setError] = React.useState<boolean>(false);
+
+  const resetFeedback = () => {
+    setFeedback("");
+    setOldPasswordFeedback({ isInvalid: false, errorMessage: "" });
+    setNewPasswordFeedback({ isInvalid: false, errorMessage: "" });
+    setConfirmPasswordFeedback({ isInvalid: false, errorMessage: "" });
+  };
+
+  const resetForm = () => {
+    setOldPassword("");
+    setNewPassword("");
+    setConfirmPassword("");
+    setSubmitted(false);
+    setError(false);
+    resetFeedback();
+  };
+
+  const submit = () => {
+    resetFeedback();
+
+    if (oldPassword === "" || newPassword === "" || confirmPassword === "") {
+      setOldPasswordFeedback({ isInvalid: true, errorMessage: "Required" });
+      setNewPasswordFeedback({ isInvalid: true, errorMessage: "Required" });
+      setConfirmPasswordFeedback({ isInvalid: true, errorMessage: "Required" });
+      setFeedback("Please enter all fields.");
+    } else if (newPassword !== confirmPassword) {
+      setNewPasswordFeedback({ isInvalid: true, errorMessage: "" });
+      setConfirmPasswordFeedback({
+        isInvalid: true,
+        errorMessage: "Passwords do not match.",
+      });
+    } else if (oldPassword === newPassword) {
+      setOldPasswordFeedback({ isInvalid: true, errorMessage: "" });
+      setNewPasswordFeedback({
+        isInvalid: true,
+        errorMessage: "New password and old password must be different.",
+      });
+    } else {
+      const auth = getAuth(firebaseApp);
+
+      if (authenticatedUser?.email) {
+        // Sign in on the frontend side to validate password
+        signInWithEmailAndPassword(auth, authenticatedUser.email, oldPassword)
+          .then((userCredential) => {
+            // Signed in
+            const { user } = userCredential;
+            updatePassword(user, newPassword)
+              .then(() => {
+                // Update successful
+                setSubmitted(true);
```

Comment:
Right now, when a user changes their password, they become unauthorized to make any requests. Instead, we need to log them in again programatically so that they can be authenticated again. We can do this by calling the authAPIClient.login() function which will get a new auth token from the backend and set that to local storage